shaman/controllers/api/builds/projects.py

```diff
@@ -33,22 +33,38 @@ def index_get(self):
     def index_post(self):
         if not self.project:
             self.project = models.get_or_create(Project, name=self.project_name)
-        build_url = request.json["build_url"]
-        repo = Build.query.filter_by(build_url=build_url).first()
-        if not repo:
-            data = dict(
-                project=self.project,
-                ref=request.json["ref"],
-                sha1=request.json["sha1"],
-                flavor=request.json["flavor"],
-            )
-            repo = models.get_or_create(Build, **data)
-        update_data = dict(
-            status=request.json["status"],
-            url=request.json.get("url", ""),
-            extra=request.json.get("extra", dict()),
+        data = dict(
+            project=self.project,
+            ref=request.json["ref"],
+            sha1=request.json["sha1"],
+            flavor=request.json.get("flavor", "default"),
+            extra=request.json.get('extra', dict()),
+            distro=request.json.get('distro'),
+            distro_version=request.json.get('distro_version'),
+            url=request.json.get("url"),
+            log_url=request.json.get("log_url"),
+            build_id=request.json.get("build_id"),
+            status=request.json.get("status"),
+            distro_arch=request.json.get("distro_arch"),
+        )
+        models.get_or_create(Build, **data)
+        return {}
+
+    #TODO: we need schema validation on this method
+    @secure(basic_auth)
+    @index.when(method='PUT', template='json')
+    def index_put(self, **kw):
+        data = dict(
+            url=kw.get("url"),
+        )
+        build = Build.query.filter_by(url=kw.get("url")).first()
+        if not build:
+            abort(404)
+        data = dict(
+            status=kw.get("status"),
+            completed=kw.get("completed"),
```

Comment:
from the model, it looks like `completed` is a `DateTime` field, but this looks like it will get a string? If it expects a datetime object, Jenkins is not providing one.